app/services/charges/charge_models/percentage_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Charges
+  module ChargeModels
+    class PercentageService < Charges::ChargeModels::BaseService
+      protected
+
+      def compute_amount(value)
+        compute_percentage_amount(value) + compute_fixed_amount(value)
+      end
+
+      def compute_percentage_amount(value)
+        value * rate
+      end
+
+      def compute_fixed_amount(value)
+        return 0 if fixed_amount.zero?
```

Comment:
You can keep it, but actually if `fixed_amount` is zero, the method will return 0 in both cases :)

Reply:
🤦 removed in other PR!